modules/core/arrow-test/src/main/kotlin/arrow/test/laws/ApplicativeLaws.kt

```diff
@@ -49,13 +66,33 @@ object ApplicativeLaws {
       FF.run { fa.map(f) }.equalUnderTheLaw(fa.ap(just(f)), EQ)
     }
 
-  fun <F> Applicative<F>.cartesianBuilderMap(EQ: Eq<Kind<F, Int>>): Unit =
-    forAll(Gen.intSmall(), Gen.intSmall(), Gen.intSmall(), Gen.intSmall(), Gen.intSmall(), Gen.intSmall()) { a: Int, b: Int, c: Int, d: Int, e: Int, f: Int ->
-      mapN(just(a), just(b), just(c), just(d), just(e), just(f)) { (x, y, z, u, v, w) -> x + y + z - u - v - w }.equalUnderTheLaw(just(a + b + c - d - e - f), EQ)
+  fun <F> Applicative<F>.cartesianBuilderMap(EQ: Eq<Kind<F, Tuple3<Int, Int, Int>>>): Unit =
+    forAll(Gen.intSmall(), Gen.intSmall(), Gen.intSmall()) { a: Int, b: Int, c: Int ->
+      mapN(just(a), just(b), just(c)) { it }.equalUnderTheLaw(just(Tuple3(a, b, c)), EQ)
+    }
+
+  fun <F> Applicative<F>.cartesianBuilderTupled2(EQ: Eq<Kind<F, Tuple2<Int, Int>>>): Unit =
+    forAll(Gen.intSmall(), Gen.intSmall()) { a: Int, b: Int ->
+      tupledN(just(a), just(b)).equalUnderTheLaw(just(Tuple2(a, b)), EQ)
+    }
+
+  fun <F> Applicative<F>.cartesianBuilderTupled3(EQ: Eq<Kind<F, Tuple3<Int, Int, Int>>>): Unit =
+    forAll(Gen.intSmall(), Gen.intSmall(), Gen.intSmall()) { a: Int, b: Int, c: Int ->
+      tupledN(just(a), just(b), just(c)).equalUnderTheLaw(just(Tuple3(a, b, c)), EQ)
+    }
+
+  fun <F> Applicative<F>.replicateSize(EQ: Eq<Kind<F, Int>>): Unit =
+    forAll(Gen.intSmall()) { n ->
```

Comment:
```suggestion
    forAll(Gen.choose(0, 100)) { n ->
```

I looked into `Gen.intSmall` and it's defined for values that would still `StackOverflow` which is  something we do not protect from in `Aplicative` so we should use `Gen.choose(0, 100)` instead for `replicate`.